src/core/Templater.ts

```diff
@@ -399,35 +399,53 @@ export class Templater {
             return;
         }
 
-        // TODO: find a better way to do this
-        // Currently, I have to wait for the daily note plugin to add the file content before replacing
-        // Not a problem with Calendar however since it creates the file with the existing content
-        await delay(300);
-
-        if (
-            file.stat.size == 0 &&
-            templater.plugin.settings.enable_folder_templates
-        ) {
-            const folder_template_match =
-                templater.get_new_file_template_for_folder(file.parent);
-            if (!folder_template_match) {
-                return;
-            }
+        await new Promise<void>((resolve) => {
+            let timeout: NodeJS.Timeout;
```

Comment:
`NodeJS.Timeout` should just be `Timeout` since we are running in a browser context and not a node context

Reply:
Nice catch, though [it seems it should just be `number`](https://github.com/microsoft/TypeScript/blob/12ab0fea9f45c48ea644c8273bedde235c8bcc61/lib/lib.dom.d.ts#L18048) in the browser context.

Reply:
P.s. do you know if there is a good reason `@types/node` is included in devDeps? It's confusing my editor unless I explicitly prefix as `window.setTimeout` / `window.clearTimeout`.

Reply:
The Obsidian API definition file actually [recommends](https://github.com/obsidianmd/obsidian-api/blob/6b2138aa28fb6d6f58f78e6fabec02186fdc0f82/obsidian.d.ts#L666) using `window.` in a similar case to avoid TS getting confused:

```ts
Use {@link window.setInterval} instead of {@link setInterval} to avoid TypeScript confusing between NodeJS vs Browser API
```

Reply:
The typings in this plugin are really messed up. I opened a PR today to fix them. In other plugins I've used `Timeout` without issue. `window.` is fine too.